src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/UpToDateCheckImplicitConfiguredInput.cs

```diff
@@ -659,15 +659,30 @@ ProjectCopyData UpdateCopyData()
                             .Select(pair => new CopyItem(path: pair.Key, metadata: pair.Value))
                             .ToImmutableArray();
 
-                        // Exclude any project references for which we do not reference the output assembly.
-                        // This avoids following references to certain "private" project references, such as PrivateP2PCaching.proj in NBGV.
-                        ImmutableArray<string> referenceItems = change2.After.Items.Where(pair => pair.Value.GetBoolProperty(ResolvedProjectReference.ReferenceOutputAssemblyProperty) == true).Select(item => item.Key).ToImmutableArray();
+                        ImmutableArray<string> referenceItems = change2.After.Items.Where(pair => IncludeProjectReference(pair.Value)).Select(item => item.Key).ToImmutableArray();
 
                         return new ProjectCopyData(msBuildProjectFullPath, targetPath, copyItems, referenceItems);
                     }
                 }
 
                 return ProjectCopyData;
+
+                static bool IncludeProjectReference(IImmutableDictionary<string, string> metadata)
+                {
+                    // Exclude any project references for which we do not reference the output assembly.
+                    if (metadata.GetBoolProperty(ResolvedProjectReference.ReferenceOutputAssemblyProperty) == false)
+                    {
+                        return false;
+                    }
+
+                    // Exclude any project references having Private="false" (aka CopyLocal="No").
+                    if (metadata.GetBoolProperty(ResolvedProjectReference.PrivateProperty) == false)
+                    {
+                        return false;
+                    }
+
+                    return true;
```

Comment:
Was it an intentional change to invert the logic for `ReferenceOutputAssemblyProperty`? Meaning before, only `true` items were included but now `null` and `true` are included.

Additionally, subsequent boolean logic gets harder for me to understand when it is using statement flow to denote truthy/falsey-ness over booleans actually interacting with each other directly. Meaning, these separate statements equate directly to this one expression:
```suggestion
                    // Exclude any project references for which we do not reference the output assembly, and
                    // exclude any project references having Private="false" (aka CopyLocal="No").
                    return !(metadata.GetBoolProperty(ResolvedProjectReference.ReferenceOutputAssemblyProperty) is false ||
                             metadata.GetBoolProperty(ResolvedProjectReference.PrivateProperty) is false);
```

Reply:
> Was it an intentional change

Yes, thanks. Good spot. If the value is unspecified (which I haven't seen in practice) then we assume the assembly is referenced.

> gets harder for me to understand

This is one of those style things where we probably won't see eye to eye. I used to compress code a lot, but now find it more readable to spread things out a bit. Especially when I'm reading other people's code. In your suggested code, it takes me a while to parse `!(A || B)`, especially as it is split across multiple lines and the `||` is off screen.

I also like putting the comments directly with the clauses they apply to.

Since your review I've added a third clause.

As mentioned in the newly added comment, this logic will require some further tweaking. This method may go away anyway.

Right now I just want to unblock dogfooders who are hitting these errors.

Reply:
Yeah. It is one of those preference things. For me, it is easier to understand things as expressions because I internally convert things to expressions. So, if they are already written that way, it is less work for me to understand. 😁

I've been trying to figure this out for a long time, but I think I do this process:
1. Convert code into shapes
2. Identify if there are any patterns in the shapes
3. If the patterns aren't something I immediately understand, I have to start reading line-by-line and construct a simplified metal expression of the code
4. Lastly, I mentally test input/output from the expression to see if it satisfies the code's intent

For me, the shorter (meaning, less tall) the shapes are, the easier it is for me to see the patterns. I think it is a matter of scale, like how you might see the pattern on the face of building's architectural design, but then it may be harder to realize that same pattern exists throughout many buildings in an entire city. And don't ask me why I have to do a 4-stage process to understand code, but it just kinda developed that way. 🙃